apps/rewards/app/components/Panel/NewReward/NewReward.js

```diff
@@ -172,31 +174,82 @@ class NewRewardClass extends React.Component {
 
   setSemanticErrors = (changed) => {
+
     const state = { ...this.state, ...changed }
     const semanticErrors = []
     const warnings = []
+
     if (state.referenceAsset === OTHER && !state.customToken.isVerified)
-      semanticErrors.push('customTokenInvalid')
+      semanticErrors.push({
+        name: 'customTokenInvalid'
+      })
     if (state.rewardType === ONE_TIME_MERIT && state.transferable)
-      semanticErrors.push('meritTokenTransferable')
+      semanticErrors.push({
+        name: 'meritTokenTransferable'
+      })
     if (toWei(state.amount) > +state.amountToken.amount)
-      semanticErrors.push('amountOverBalance')
-    if (state.rewardType === ONE_TIME_DIVIDEND &&
-        moment(state.dateReference).isBefore(tomorrow, 'day'))
-      semanticErrors.push('dateReferencePassed')
+      semanticErrors.push({
+        name: 'amountOverBalance'
+      })
     if (state.rewardType === RECURRING_DIVIDEND ||
         state.rewardType === ONE_TIME_MERIT) {
-      const start = moment(state.dateStart), end = moment(state.dateEnd)
-      if (start.isBefore(tomorrow, 'day'))
-        semanticErrors.push('dateStartPassed')
-      if (end.isBefore(tomorrow, 'day'))
-        semanticErrors.push('dateEndPassed')
-      if (start.isAfter(end, 'day'))
-        semanticErrors.push('dateStartAfterEnd')
+      if (isBefore(state.dateEnd, state.dateStart))
+        semanticErrors.push({
+          name: 'dateStartAfterEnd'
+        })
     }
+    if (state.referenceAsset !== null && state.rewardType !== null) {
+      let tokenCreationDate, tokenSymbol
+      if (state.referenceAsset === OTHER && state.customToken.isVerified) {
+        tokenCreationDate = state.customToken.creationDate
+        tokenSymbol = state.customToken.symbol
+      }
+      else {
+        const refToken = this.props.refTokens.find(t => {
+          return t.address === state.referenceAsset.props.address
+        })
+        tokenCreationDate = refToken.creationDate
+        tokenSymbol = refToken.symbol
+      }
+      if (state.rewardType === RECURRING_DIVIDEND ||
+          state.rewardType === ONE_TIME_MERIT) {
+        if (isBefore(state.dateStart, tokenCreationDate)) {
+          semanticErrors.push({
+            name: 'dateBeforeAsset',
+            params: {
+              dateType: 'start',
+              tokenSymbol: tokenSymbol,
+            }
+          })
+        }
+        if (isBefore(state.dateEnd, tokenCreationDate)) {
+          semanticErrors.push({
+            name: 'dateBeforeAsset',
+            params: {
+              dateType: 'end',
+              tokenSymbol: tokenSymbol,
+            }
+          })
+        }
+      }
+      if (state.rewardType === ONE_TIME_DIVIDEND
+          && isBefore(state.dateReference, tokenCreationDate)) {
+        semanticErrors.push({
+          name: 'dateBeforeAsset',
+          params: {
+            dateType: 'reference',
+            tokenSymbol: tokenSymbol,
+          }
+        })
+      }
+    }
```

Comment:
We don't necessarily need to change anything for now, but I find all of this deeply-nested logic impossible to follow. 

* if the reference asset is not null and reward type is not null
  * if reference asset is OTHER and customToken is verified
    * use the symbol and creationDate from the customToken
  * else (if reference asset is OTHER but customToken is unverified, or if reference asset is anything else)
    * use the symbol and creationDate from the reference token that has the same address as the reference asset
  * furthermore, if rewardType is recurring dividend or one-item merit
    * if start date is before creationDate, add error for start date
    * if end date is before creationDate, add error for end date
  * finally, if rewardType is one-time dividend and date reference is before creationDate, add error for dateReference

Phew! Did I get that right?

In my experience, these sorts of complicated logic trees almost always have hidden bugs, and generate new bugs frequently. 

Again, I'm not sure how to fix it right now, but I want to note this complexity, so we can start thinking about how we might be able to restructure this component to allow simplified logic here.

Reply:
I don't think you got it right, which proves your point :)

Let me rework this with your suggestions. I have a few ideas.

Reply:
OK, done. Let me know what you think, @chadoh